src/ng/location.js

```diff
@@ -93,7 +94,7 @@ function stripHash(url) {
 }
 
 function trimEmptyHash(url) {
```

Comment:
Can we remove some calls to this within this file now that `$browser` does it?

Reply:
[Absolutely!][1]

> Note that is many places we trim the empty hash before using the returned url. (If we decide this fix is desired, I will remove those unnecessary calls.)

[1]: https://github.com/angular/angular.js/pull/16636#issuecomment-406356498